lib/python/rose/bush.py

```diff
@@ -518,6 +519,19 @@ def view(self, user, suite, path, path_in_tar=None, mode=None):
         else:
             file_content = self.suite_engine_proc.is_conf(path)
         template = self.template_env.get_template("view.html")
+
+        # convert url(s) to hyperlinks if in text mode
+        if mode == None:
+            anchor = lambda url: '<a href="%s">%s</a>' % (url, url)
+            urlregex = '(http|https|ftp)://[^\s]+'
```

Comment:
You may want to use `re.compile` to pre-compile the regular expression to speed things up.